viewing_party/party.py

```diff
@@ -0,0 +1,145 @@
+    for item in amandas_data["friends"][1]["watched"]:
+        if item not in amandas_friends_watched:
+            amandas_friends_watched.append(item)
+    
+    for item in amandas_watched:
+        if item not in amandas_friends_watched:
+            unique_watched.append(item)
+
+    return unique_watched
+    
+def get_friends_unique_watched(amandas_data):
+    unique_watched = []
+    amandas_watched = amandas_data["watched"]
+    amandas_friends_watched =amandas_data["friends"][0]["watched"]
+    for item in amandas_data["friends"][1]["watched"]:
+        if item not in amandas_friends_watched:
+            amandas_friends_watched.append(item)
+    
+    for item in amandas_friends_watched:
+        if item not in amandas_watched:
+            unique_watched.append(item)
+
+    return unique_watched
+
+
+def get_available_recs(amandas_data):
+    friends_unique_movie = []
+    temp_unssen_movie_amanda= []
+    rec_movie = []
+    unseen_amanda_movie_title = []
+    
+    # find the unique movies watched by friends if the host is found in the Amanda subscriptions. 
+    for item in amandas_data["friends"][0]["watched"]:
+        if item["host"] in amandas_data["subscriptions"] :
+            friends_unique_movie.append(item)
+            
+    for item in amandas_data["friends"][1]["watched"]:
+        if item["host"] in amandas_data["subscriptions"]  and item not in friends_unique_movie:
+            friends_unique_movie.append(item)
+    # find the unique movie title watchd by friends with out the host       
+    new_friends_unique_movie_title = [{k:v for k,v in item.items() if k != "host"} for item in friends_unique_movie ]
+    
+    
+    # find the movie title unseen by Amanda as a dictionary and append to a list
+    for item in new_friends_unique_movie_title:      
+        if item not in amandas_data["watched"]:
+            temp_unssen_movie_amanda.append(item)
+    
+    # find the movie unseen by amanda and add to a list
+    for item in temp_unssen_movie_amanda:
+        for value in item.values():
+            unseen_amanda_movie_title.append(value)
+    
+    # recommend movie title with the host for Amanda and put in a list
+    for item in friends_unique_movie:
+        if item["title"] in unseen_amanda_movie_title:
+            rec_movie.append(item)
+
+    return rec_movie
+
+def get_new_rec_by_genre(sonyas_data):
+    rec_unique_movie = []
+    if sonyas_data["watched"] == [] or (sonyas_data["friends"][0]["watched"]
+                and sonyas_data["friends"][1]["watched"]) == []:
+        return []
+
+    for item in sonyas_data["friends"][0]["watched"]:
+        if item not in sonyas_data["watched"] :
+            rec_unique_movie.append(item)
+            
+    for item in sonyas_data["friends"][1]["watched"]:
+        if item not in sonyas_data["watched"] and item not in rec_unique_movie:
+            rec_unique_movie.append(item)
+            
+    return rec_unique_movie    
```

Comment:
There is a logical bug in this function. This function does not check for the most popular genre or do any checking against the genre. It will produce the same output as 'get_friends_unique_watched'.